scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -569,11 +569,14 @@ def handle(self):
                 try:
                     (stdout, stderr) = proc.communicate(timeout=60)
                     # ignore unencodable unicode chars
-                    logger.debug(stdout.decode(errors = "ignore"))
+                    decoded_stdout = stdout.decode(errors="ignore")
+                    decoded_stderr = stderr.decode(errors="ignore")
+                    logger.debug("Flash command stdout: %s", decoded_stdout)
+                    logger.error("Flash command stderr: %s", decoded_stderr)
 
                     if proc.returncode != 0:
                         self.instance.status = "error"
-                        self.instance.reason = "Device issue (Flash error?)"
+                        self.instance.reason = f'Device issue (Flash error?): {decoded_stderr}. returncode: {proc.returncode}'
                         flash_error = True
                         with open(d_log, "w") as dlog_fp:
                             dlog_fp.write(stderr.decode())
```

Comment:
you can use here `decoded_stderr`: `dlog_fp.write(decoded_stderr)`

Reply:
I think the idea behind the current implementation is to keep `stderr` in `d_log` file as it is too big for a json string field.
It would be good to add only the return code here.